app/uk/gov/hmrc/gform/eval/smartstring/SmartStringEvaluatorFactory.scala

```diff
@@ -106,8 +106,8 @@ class RealSmartStringEvaluatorFactory() extends SmartStringEvaluatorFactory {
                     case IsChoice(choice) =>
                       val optionsList = choice.options.zipWithIndex
                         .map {
-                          case (OptionData.IndexBased(label, _), i)        => i.toString -> label
-                          case (OptionData.ValueBased(label, value, _), i) => value      -> label
+                          case (OptionData.IndexBased(label, _, _, _), i)        => i.toString -> label
+                          case (OptionData.ValueBased(label, _, _, _, value), i) => value      -> label
```

Comment:
`i` is not used, anywhere it can be `_`
Also in may places we keep adding fields to the case classes and we have to update the matchings.
Wouldn't it be better if we have like
                          case (o: OptionData.ValueBased, _) => o.value -> o.label

Reply:
Using typed based matching would mean to give up majority of guarantees Scala pattern matching is giving us.
I think it is one of the best features to let compiler show you all the places where you may potentially use newly added field (even if you end up not needing it to use in a single place).